src/main/repository/PromotionRepository.ts

```diff
@@ -16,7 +22,11 @@ export class PromotionRepository extends Repository<Promotion> {
       promotionQuery &&
       JSON.stringify(promotionQuery) !== JSON.stringify({})
     ) {
-      return this.applyQueryOptions(promotionQuery);
+      const promotions = await this.applyQueryOptions(promotionQuery);
+      if (promotionQuery.userId && promotions.length != 0) {
```

Comment:
Just a minor thing but I think we can condense more!

```suggestion
      if (promotionQuery.userId && promotions.length) {
```

Reply:
Good point thanks!